Source/SpectrumAnalysis/KTSeqTrackFinder.cc

```diff
@@ -0,0 +1,539 @@
+            double freqMin = power_spectrum->GetBinLowEdge(fMinBin);
+            double freqMax = power_spectrum->GetBinLowEdge(fMaxBin) + power_spectrum->GetBinWidth();
+            KTSpline::Implementation* splineImp = spline->Implement(nBins, freqMin, freqMax);
+
+
+            fBinWidth = power_spectrum->GetBinWidth();
+
+            // I need a deepcopy of the amplitudes of the power spectrum
+            // As I am looping over each bin anyway I will fill the vector "slice" at the same time
+            std::vector< double > slice(fMaxBin +1);
+            
+            // filling it with zeros first shouldn't be necessary
+            //std::fill(slice.begin(), slice.end(), 0.0);
+
+            double new_TimeInAcq = slHeader.GetTimeInAcq() + 0.5 * slHeader.GetSliceLength();
+            double new_TimeInRunC = slHeader.GetTimeInRun() + 0.5 * slHeader.GetSliceLength();
+            KTDEBUG(stflog, "new_TimeInAcq is " << new_TimeInAcq);
+
+
+            // this vector will collect the high power bins in this power slice
+            std::vector<Point> Points;
+
+            // the trimming limits are the average of the entire slice power
+            // trimming_factor x trimming_limits are the threshold for the line edges
+            // this is part of dans algorithm but seems to only make sense with a flat background
+            // ->replaced by point threshold
+            // double new_TrimmingLimits;
+            double value;
+
+
+            // loop over bins, checking against the threshold
+            KTDEBUG(stflog, "Collecting high power bins");
+            for (unsigned iBin=fMinBin; iBin<=fMaxBin; ++iBin)
+            {
+                value = (*power_spectrum)(iBin);
+                slice.at(iBin) = value;
+                //new_TrimmingLimits += value;
+                double threshold = fSNRPowerThreshold * (*splineImp)(iBin - fMinBin);
+
+                // currently this is the only implemented mode
+                if (fMode == eSNR_Power and value >= threshold)
+                {
+                    double PointFreq = fBinWidth * ((double)iBin + 0.5);
+                    Point new_Point(iBin, PointFreq, new_TimeInAcq, new_TimeInRunC, value, threshold, AcqID, iComponent);
+                    Points.push_back(new_Point);
+                    //KTDEBUG(stflog, "Collected track point candidate");
+                }
+            }
+
+            //new_TrimmingLimits = new_TrimmingLimits/nBins;
+
+            // I want to iterate over the collected points in order of descending power
+            // sort points by power
+            std::sort(Points.begin(), Points.end(),std::less<Point>());
+
+            // Loop over the high power points
+            this->LoopOverHighPowerPoints(slice, Points, iComponent);
+
+        }
+        return true;
+    }
+
+
+    bool KTSeqTrackFinder::LoopOverHighPowerPoints(std::vector<double>& slice, std::vector<Point>& Points, unsigned component)
```

Comment:
1. Please switch suffix operators (e.g. `iterator++``) to prefix operators (e.g. `++iterator`) for improved efficiency.

1. There are a few places where the indentation has extra spaces.

1. Variable names should start with lower-case letters.  A few need fixing: `Points`, `PointIt`, `LineIt`, `Freq`, `Amplitude`